test/e2e/allocate.go

```diff
@@ -0,0 +1,289 @@
+package e2e
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	schedulingv1beta1 "volcano.sh/volcano/pkg/apis/scheduling/v1beta1"
+)
+
+var _ = Describe("Job E2E Test", func() {
+	It("schedule job when resource is enough", func() {
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2000m"),
+				corev1.ResourceMemory: resource.MustParse("2048Mi")},
+		})
+		defer cleanupTestContext(ctx)
+
+		slot1 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("500m"),
+			corev1.ResourceMemory: resource.MustParse("512Mi")}
+		slot2 := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1000m"),
+			corev1.ResourceMemory: resource.MustParse("1024Mi")}
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot1,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "low"
+		lowReqJob := createJob(ctx, job)
+		err := waitJobReady(ctx, lowReqJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "high"
+		job.tasks[0].req = slot2
+		highReqJob := createJob(ctx, job)
+		err = waitJobReady(ctx, highReqJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = compareNodename(ctx, lowReqJob, highReqJob)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("schedule job when resource is NOT enough", func() {
+		ctx := initTestContext(options{
+			nodesNumLimit: 2,
+			nodesResourceLimit: corev1.ResourceList{
```

Comment:
does this mean the resource specified is idle?

Reply:
This means the resources specified is required(Prepare form cluster nodes).